compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1085,16 +1023,21 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     );
                     if let Some(binding) = binding {
                         match binding.kind {
-                            ConvertedBindingKind::Equality(term) => {
+                            hir::TypeBindingKind::Equality { term } => {
+                                let term: ty::Term<'_> = match term {
+                                    hir::Term::Ty(ty) => self.ast_ty_to_ty(ty).into(),
+                                    hir::Term::Const(ct) => {
+                                        ty::Const::from_anon_const(tcx, ct.def_id).into()
```

Comment:
This might leak `{type error}` since we haven't actually resolved & feed the type to `type_of(AnonConst)` if we reach this case. The pretty-printer doesn't seem to care though (not sure if it can actually crash on certain inputs though).

Alternatively, I could pretty-print the HIR of the const but I'd need to create a public helper, `const_to_string`, since `rustc_hir_pretty` doesn't seem to expose anything like that yet?